### DISCUSSION TITLE
k8s 部署，ingress是GRPC HTTP2协议，如何开启TLS配置，默认证书有效期是多少,JAVA SDK支持TLS链接吗

### DISCUSSION THREAD

**Discussion by herozeng** [milvus-io/milvus#24308] [Q&A and General discussion]
通过helm部署了milvus, 暴漏的ingress 地址，通过JAVA SDK链接不上，可能是没开启TLS配置的原因

**Comment by herozeng** [comment]
from pymilvus import connections

_HOST = '127.0.0.1'
_PORT = '19530'

print(f"\nCreate connection...")
connections.connect(host=_HOST, port=_PORT, secure=True, client_pem_path="cert/client.pem",
                        client_key_path="cert/client.key",
                        ca_pem_path="cert/ca.pem", server_name="localhost")
print(f"\nList connections:")
print(connections.list_connections())


只看到python 证书SDk的配置,java SDk没有

**Reply by yhmo** [reply]
JAVA SDK doesn't support TLS certification.

**Reply by cliu701** [reply]
Is there a plan to support tls for java sdk?

**Comment by yhmo** [comment]
> Is there a plan to support tls for java sdk?

Yes, we do want to implement it. But quite busy recently for high priority tasks, will do it once the important tasks are done.
